pkg/autoscaler/autoscaler.go

```diff
@@ -196,6 +200,27 @@ func (a *Autoscaler) Scale(ctx context.Context, now time.Time) (int32, bool) {
 	desiredStablePodCount := a.podCountLimited(math.Ceil(observedStableConcurrency/target), readyPodsCount)
 	desiredPanicPodCount := a.podCountLimited(math.Ceil(observedPanicConcurrency/target), readyPodsCount)
 
+	//// Do nothing when we have no data.
+	if len(a.bucketed) == 0 {
```

Comment:
I am wondering if it is necessary to check this again as we checked this at [line 189](https://github.com/knative/serving/pull/3579/files#diff-a6c4792dc41958d712d5dc9615145d28R189). And looks like the `bucketed` won't be changed after that line.